catalyst-gateway/Earthfile

```diff
@@ -47,29 +48,29 @@ all-hosts-build:
     BUILD --platform=linux/amd64 --platform=linux/arm64 +build
 
 package-cat-gateway:
+    ARG tag="latest"
+
     FROM alpine:3.19
+    WORKDIR /cat-gateway
+
+    RUN apk add --no-cache gcc bash
 
+    COPY +build/cat-gateway .
+    COPY ./scripts/entry.sh .
+
+    ENTRYPOINT ./entry.sh
+    SAVE IMAGE cat-gateway:$tag
+
+package-cat-gateway-with-preprod-snapshot:
```

Comment:
What do you think about having only one package-cat-gateway target and passing it a ARG to  select if we want a snapshot or not? Something like
```
package-cat-gateway:
ARG snapshot="false"
.....
.....
IF [$snapshot = "true"]
    COPY github.com/input-output-hk/catalyst-ci/earthly/mithril_snapshot:v2.11.1+package-preprod-snapshot/snapshot /tmp/preprod
END
```
Also can we package only the preprod snapshot or mainnet as well? It might be worth thinking about having a option to choose the network. So someting along this lines:
```
package-cat-gateway:
ARG snapshot="false"
ARG network="preprod"
.....
.....
IF [$snapshot = "true"]
    COPY github.com/input-output-hk/catalyst-ci/earthly/mithril_snapshot:v2.11.1+package-$network-snapshot/snapshot /tmp/$network
END
```

Reply:
I am totally understand your intention.
But for now as we dont have such flexible downloading snapshot target I will leave as it is.